client/encoding.go

```diff
@@ -0,0 +1,34 @@
+package client
+
+import (
+	"encoding/json"
+
+	p "github.com/coniks-sys/coniks-go/protocol"
+)
+
+func UnmarshalResponse(t int, msg []byte) (
+	p.DirectoryResponse, p.ErrorCode) {
+	type Response struct {
+		Error             p.ErrorCode
+		DirectoryResponse json.RawMessage
+	}
+	var res Response
+	if err := json.Unmarshal(msg, &res); err != nil {
+		return nil, p.ErrorMalformedDirectoryMessage
+	}
```

Comment:
`DirectoryResponse` is `omitempty` for the places where we call `NewErrorResponse()`. Looks like you need a `nil` check here, but only when `!Success`.  And some tests for it, if I have this right.

We probably also need to distinguish which error codes it's ok to be getting an error response that omits the `DirectoryResponse`.